src/Tools/Source/CompilerGeneratorTools/Source/CSharpSyntaxGenerator/SourceGenerator.cs

```diff
@@ -96,10 +116,15 @@ public void Execute(GeneratorExecutionContext context)
 
             TreeFlattening.FlattenChildren(tree);
 
+            var cachedSources = ImmutableArray.CreateBuilder<(string hintName, SourceText sourceText)>();
             AddResult(writer => SourceWriter.WriteMain(writer, tree, context.CancellationToken), "Syntax.xml.Main.Generated.cs");
             AddResult(writer => SourceWriter.WriteInternal(writer, tree, context.CancellationToken), "Syntax.xml.Internal.Generated.cs");
             AddResult(writer => SourceWriter.WriteSyntax(writer, tree, context.CancellationToken), "Syntax.xml.Syntax.Generated.cs");
 
+            // Overwrite the cached result with the new result. This is an opportunistic cache, so as long as the write
+            // is atomic (which it is for a single pointer) synchronization is unnecessary.
+            s_cachedResult = Tuple.Create(currentChecksum, cachedSources.ToImmutable());
+
```

Comment:
```suggestion
            s_cachedResult = Tuple.Create(currentChecksum, cachedSources.ToImmutable());
            return;

```

Reply:
Done as part of extracting the base class.